modules/drivers/mongo/src/metabase/driver/mongo/connection.clj

```diff
@@ -33,7 +33,6 @@
      host
      (when (and (not use-srv) (some? port)) (str ":" port))
      "/"
-     dbname
```

Comment:
We are removing the the default auth db here, right? Are we sure it's set somewhere else or that none of our clients are relying on it?

Reply:
Exactly. We do set the database to authenticate against _always_, in case credentials are provided. See the [mongo/connection.clj](https://github.com/metabase/metabase/blob/435f9acbad63b434c08291d1306dd450eec80777/modules/drivers/mongo/src/metabase/driver/mongo/connection.clj#L80).

Based on [Mongo connection string docs](https://www.mongodb.com/docs/manual/reference/connection-string/#connection-string-components) (see the `username:password@` section), if the credentials are provided, `authSource` is used if set.`defaultauthdb` is used if there's no `authSource`. And if there is no `defaultauthdb`, `admin` database is used.

My understanding is that if we use `authSource`, `defaultauthdb` is redundant. I believe no one can rely on that, as `authSource` should always have a precendece.

When I was removing monger I've left the `defaultauthdb` in the connection string, because `srv-conn-str` function in [Monger removal PR](https://github.com/metabase/metabase/pull/38017/files) (removed from `mongo/util.clj`) used both. But I believe use of both parameters is redundant.